modules/rte_mis_core/src/Plugin/Block/SiteLogoSectionBlock.php

```diff
@@ -0,0 +1,148 @@
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entityTypeManager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $config = $this->getConfiguration();
+    $build = [];
+
+    $heading = isset($config['site_logo_heading']) && $config['site_logo_heading'] !== '' ? $config['site_logo_heading'] : 'Government Of India';
+    $subtext = isset($config['site_logo_subtext']) && $config['site_logo_subtext'] !== '' ? $config['site_logo_subtext'] : 'School Education Department Right to Education(RTE) Portal';
+    $logoImg = $config['site_logo_image'] ?? '';
+
+    $build['heading']['#markup'] = '<h2>' . $heading . '</h2>';
+    $build['sub_heading']['#markup'] = '<p>' . $subtext . '</p>';
+
+    if (empty($logoImg[0])) {
+      // Load and display the default image from the assets folder.
+      $defaultImage = '/profiles/contrib/rte_mis/modules/rte_mis_core/asset/img/Emblem_of_India.svg';
+      $build['image'] = [
+        '#theme' => 'image',
+        '#width' => 50,
+        '#height' => 100,
+        '#uri' => $defaultImage,
+      ];
+    }
+    else {
+      // Display the user-uploaded image.
+      $file_entity = $this->entityTypeManager->getStorage('file');
+      if ($file = $file_entity->load($logoImg[0])) {
+        $build['image'] = [
+          '#theme' => 'image',
+          '#width' => 50,
+          '#height' => 100,
+          '#uri' => $file->getFileUri(),
+        ];
+      }
+    }
+
+    return $build;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form = parent::blockForm($form, $form_state);
+
+    $config = $this->getConfiguration();
+
+    $form['site_logo_image'] = [
+      '#type' => 'managed_file',
+      '#title' => $this->t('Site Logo'),
+      '#default_value' => $config['site_logo_image'] ?? '',
+      '#upload_location' => 'public://site_logo/asset/',
+      '#upload_validators' => [
+        'file_validate_extensions' => ['jpg jpeg png svg'],
+      ],
+      '#description' => $this->t('Site Logo'),
+    ];
+
+    $form['site_logo_heading'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Site Logo Heading'),
+      '#default_value' => isset($config['site_logo_heading']) && $config['site_logo_heading'] !== '' ? $config['site_logo_heading'] : 'Government Of India',
```

Comment:
Could you keep the static string wrapped inside t()?

Reply:
done.